File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.cs
public class Solution {
    public int MaximumElementAfterDecrementingAndRearranging(int[] arr) {
        int n = arr.Length;
        Array.Sort(arr);
        int ans = 1;

        for (int i = 1; i < n; i++) {
            if (arr[i] >= ans + 1) {
                ans++;
            }
        }

        return ans;
    }
}

File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        arr.sort()
        ans = 1
        flag = False

        for num in arr:
            if flag and num >= ans + 1:
                ans += 1
            else:
                flag = True

        return ans
